src/pvrclient-argustv.h

```diff
@@ -128,6 +134,7 @@ class cPVRClientArgusTV
   P8PLATFORM::CMutex        m_ChannelCacheMutex;
   std::vector<cChannel*>   m_TVChannels; // Local TV channel cache list needed for id to guid conversion
   std::vector<cChannel*>   m_RadioChannels; // Local Radio channel cache list needed for id to guid conversion
+  std::map <int,PVR_RECORDING_ENTRY> m_RecordingsMap;
```

Comment:
will do. Pulled this from some other code i was working on as a basic concept. Knew it was going to be bloated, but figured could clean it up when confirmation of working came through.

Reply:
I'm quite sure your approach will work. It's no rocket science, isn't it.

Reply:
Actually a problem, recordingid's in PVR_RECORDING structs are actually char's and not ints like PVR_CHANNEL. Im unsure about a char[] being a key for a map, and otherwise the current key is just order of what was returned from the backend. This isnt mapped to anything within the PVR_RECORDING that is being passed to the functions, so i cant distinguish which string (url) is for which recording without the extra struct.

Any advice? just point me in the right direction and ill get it sorted, just stumped for best way.

Reply:
Maybe https://github.com/kodi-pvr/pvr.argustv/commit/ea4daac is an alternative, but I don't know if `ArgusTV::GetRecordingById(...)` does what I expect from it.

Reply:
Why don't just make a std::string from the recording id  char* and use the string as'first' for the map? I'd prefer this over @Rechi's solution as this does not need additional backend calls.

Reply:
yeah, i think thats the way ksooo. ill make the change now and squash everything